packages/manager/apps/hycu/src/pages/listing/Listing.page.tsx

```diff
@@ -1,121 +1,216 @@
-                      onClick={() => navigateToDashboard(label)}
-                    >
-                      {label}
-                    </OsdsLink>
-                  </DataGridTextCell>
-                );
-              return <DataGridTextCell>{label}</DataGridTextCell>;
-            }
-            return <div>-</div>;
-          },
-        }));
-      setColumns(newColumns);
-    } else if (isSuccess && !flattenData?.length) {
-      navigate(urls.onboarding);
-    }
-  }, [flattenData]);
-
-  if (isError) {
-    return <ErrorBanner error={error.message} />;
-  }
-
-  if (isLoading && pageIndex === 1) {
-    return (
-      <div>
-        <Loading />
-      </div>
-    );
-  }
+  const columns = useMemo(() => {
+    return [
+      {
+        id: 'name',
+        label: t('hycu_name'),
+        cell: DatagridIdCell,
+      },
+      {
+        id: 'controller_id',
+        label: t('hycu_controller_id'),
+        cell: DatagridControllerIdCell,
+      },
+      {
+        id: 'status',
+        label: t('hycu_status'),
+        cell: DatagridStatusCell,
+      },
+      {
+        id: 'commercial_name',
+        label: t('hycu_commercial_name'),
+        cell: DatagridCommercialNameCell,
+      },
+      {
+        id: 'subscribed_date',
+        label: t('hycu_subscribed_date'),
+        cell: DatagridCreatedDateCell,
+      },
+      {
+        id: 'action',
+        label: '',
+        cell: DatagridActionCell,
+      },
+    ];
+  }, []);
 
   const header = {
     title: t('title'),
   };
 
   return (
-    <BaseLayout breadcrumb={<Breadcrumb />} header={header}>
-      <React.Suspense>
-        {columns && flattenData && (
-          <Datagrid
-            columns={columns}
-            items={flattenData || []}
-            totalItems={totalCount || 0}
-            hasNextPage={hasNextPage && !isLoading}
-            onFetchNextPage={fetchNextPage}
-            sorting={sorting}
-            onSortChange={setSorting}
-          />
-        )}
-      </React.Suspense>
-    </BaseLayout>
+    <RedirectionGuard
+      isLoading={isLoading || !flattenData}
+      condition={status === 'success' && flattenData?.length === 0}
+      route={urls.onboarding}
+      isError={isError}
+      errorComponent={
+        <OsdsMessage className="mt-4" type={ODS_MESSAGE_TYPE.error}>
+          {tError('manager_error_page_default')}
+        </OsdsMessage>
+      }
+    >
+      <BaseLayout header={header}>
+        <React.Suspense>
+          <div className="flex flex-col gap-4">
+            <div>
+              <OsdsButton
+                color={ODS_THEME_COLOR_INTENT.primary}
+                variant={ODS_BUTTON_VARIANT.stroked}
+                size={ODS_BUTTON_SIZE.sm}
+                inline
+              >
+                {t('hycu_order')}
+              </OsdsButton>
+            </div>
+            {columns && flattenData && (
+              <Datagrid
+                columns={columns}
+                items={flattenData || []}
```

Comment:
`flattenData` can't be null here
```suggestion
                items={flattenData}
```